src/lib.rs

```diff
@@ -60,7 +60,7 @@ macro_rules! asserted_const_with_type {
 #[cfg(any(apple, bsd))]
 mod bsd;
 
-#[cfg(target_os = "linux")]
+#[cfg(any(target_os = "linux", target_os = "android"))]
```

Comment:
Note that `target_os = "linux"` does not include `android`. See https://doc.rust-lang.org/reference/conditional-compilation.html#target_os .